framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp

```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
     return status;
 }
 
+int cfg_uci_get_wireless_from_ifname(enum paramType type, char *interface_name, const char param[],
+                                     char *value)
+{
+    struct uci_context *ctx = uci_alloc_context();
+    if (!ctx) {
+        ERROR("%s, uci alloc context failed!\n", __FUNCTION__);
+        return RETURN_ERR;
+    }
+
+    char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
+    struct uci_ptr ptr;
+    if ((uci_lookup_ptr(ctx, &ptr, lookup_str, true) != UCI_OK)) {
+        ERROR("%s, uci lookup failed!\n", __FUNCTION__);
+        uci_free_context(ctx);
+        return RETURN_ERR;
+    }
+
+    if (!ptr.p) {
+        ERROR("%s, returned pointer is null\n", __FUNCTION__);
+        return RETURN_ERR;
+    }
+
+    bool is_section_found = false;
+    struct uci_package *p = ptr.p;
+    struct uci_element *e = nullptr;
+    struct uci_element *n = nullptr;
+    struct uci_section *s = nullptr;
+    // Iterate over all wireless sections in the UCI DB
+    uci_foreach_element(&p->sections, e)
+    {
+        s = uci_to_section(e);
+
+        if (strncmp(s->type, "wifi-iface", MAX_UCI_BUF_LEN))
+            continue;
+
+        // Iterate over all the options in the section
+        uci_foreach_element(&s->options, n)
+        {
+            struct uci_option *o = uci_to_option(n);
+
+            if (o->type != UCI_TYPE_STRING)
+                continue;
+
+            if (strncmp(n->name, "ifname", MAX_UCI_BUF_LEN))
+                continue;
+
+            if (strncmp(interface_name, o->v.string, MAX_UCI_BUF_LEN))
+                continue;
+
+            // We reached the section containing the requested ifname
+            is_section_found = true;
+            break;
+        }
+
+        if (is_section_found) {
+            break;
+        }
+    }
+
+    if (type == TYPE_RADIO) {
```

Comment:
I really don't like the pattern
```
if (arg == 1) {
   ...
   return x;
}

// arg == 2
...
return y;
```

I.e., rely on the return for something that is a choice. The pattern is excellent for handling errors, but not for handling two alternative code paths IMO.

In this case, I much prefer either:
```
if (arg == 1) {
   ... foo = ...
} else { // arg == 2
   ... foo = ...
}
return foo;
```

or even

```
switch (arg) {
case 1:
   ...
   return x;
case 2:
   ...
   return y;
default:
   ERROR
   return z;
}
```

(nitpick)

Reply:
Kudos on giving such an elaborate comment even though you mark it as (nitpick)

Reply:
the battle between the forces of indentation and forces of if-else continues :-)
I will change it to if-else